numpy/core/function_base.py

```diff
@@ -1,10 +1,24 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
+def _index_deprecate(i, stacklevel=2):
```

Comment:
Yeah, I thought it was a good idea. Of course, it basically only makes sense if we also use it elsewhere, I somewhat expected we have enough other places. Tthough, of course it is also possible that nobody will bother to do similar fixes elsewhere for a long while....

I like using try/except. Try except seems nice to me, since it tries to use the final version we would like to be using at some point. If the final version wasn't what we try first, I would agree that the check might be better (though it also safes me from worrying about weird corner cases).